crates/egui-winit/src/clipboard.rs

```diff
@@ -132,15 +136,20 @@ fn init_arboard() -> Option<arboard::Clipboard> {
     ),
     feature = "smithay-clipboard"
 ))]
-fn init_smithay_clipboard(
-    wayland_display: Option<*mut c_void>,
+fn init_smithay_clipboard<T>(
+    _event_loop: &EventLoopWindowTarget<T>,
 ) -> Option<smithay_clipboard::Clipboard> {
-    if let Some(display) = wayland_display {
-        tracing::debug!("Initializing smithay clipboard…");
-        #[allow(unsafe_code)]
-        Some(unsafe { smithay_clipboard::Clipboard::new(display) })
-    } else {
-        tracing::debug!("Cannot initialize smithay clipboard without a display handle");
-        None
+    // Note: ideally "smithay-clipboard" would imply "wayland", but it doesn't.
+    #[cfg(feature = "wayland")]
+    {
+        use winit::platform::wayland::EventLoopWindowTargetExtWayland as _;
+        if let Some(display) = _event_loop.wayland_display() {
+            tracing::debug!("Initializing smithay clipboard…");
+            #[allow(unsafe_code)]
+            return Some(unsafe { smithay_clipboard::Clipboard::new(display) });
+        }
     }
+
+    tracing::debug!("Cannot initialize smithay clipboard without a display handle");
```

Comment:
maybe this makes more sense?

```suggestion
        } else {
            tracing::debug!("Cannot initialize smithay clipboard without a display handle");
            return;
        }
    }
```